src/NuGet.Services.EndToEnd/Support/TestSettings.cs

```diff
@@ -35,6 +35,7 @@ public enum Mode
             IReadOnlyList<string> trustedHttpsCertificates,
             string apiKey,
             string searchBaseUrl,
+            string autoCompleteUrl,
```

Comment:
In principle I totally agree with you. 

However when we want to run the search service against the staging slot, all we know is the base URL (by using the Azure management APIs to get a cloud service base URL). In that case we have to build the URL anyways. For these reason I unified the cases where a) we get the base URLs from v3/index.json and b) we get the staging base URL from Azure.

See https://github.com/NuGet/NuGet.Services.EndToEnd/blob/1cb7b82ad1932674a731e719912fd94ed1e13016/src/NuGet.Services.EndToEnd/Support/Clients/V2V3SearchClient.cs#L28-L48.

When the `searchBaseUrl` is configured in an E2E test run that means we are testing a staging slot of search service instead of the publicly visible search services. When will the `autoCompleteUrl` be specifified?

Also, nit, `autocompleteUrl` instead of `autoCompleteUrl` to align with the casing in v3/index.json.